src/api/java/com/minecolonies/api/crafting/RecipeStorage.java

```diff
@@ -228,13 +340,20 @@ public int hashCode()
     private boolean checkForFreeSpace(final List<IItemHandler> handlers)
     {
         final List<ItemStack> secondaryStacks = new ArrayList<>();
-        for (final ItemStack stack : input)
+        if(!secondaryOutputs.isEmpty())
+        {
+            secondaryStacks.addAll(secondaryOutputs);
+        }
+        else
         {
-            final ItemStack container = stack.getItem().getContainerItem(stack);
-            if (!ItemStackUtils.isEmpty(container))
+            for (final ItemStack stack : input)
             {
-                container.setCount(stack.getCount());
-                secondaryStacks.add(container);
+                final ItemStack container = stack.getItem().getContainerItem(stack);
+                if (!ItemStackUtils.isEmpty(container))
+                {
+                    container.setCount(stack.getCount());
```

Comment:
This does not work........
The API clearly states that the container Item is returned when the stack is consumed! You are not allowed to modify this stack willy nilly since that violates that contract. There might be mods out there which consume an entire stack and only return 1 container item!

Reply:
Either way this does not work, since we operate off of cleaned inputs. In the case of cake, the stack is 2 milk buckets, and getContainerItem returns one bucket, but vanilla crafting returns 2. The good news is that this code is legacy/compat. With the secondary items list, we return what the recipe says gets returned.

Reply:
If this is legacy code. Why was it added in your PR?

Reply:
The container case becomes legacy code with this PR, there was a bug reported around producing too few buckets while I was working on it, so I included it to avoid more merge conflicts.